Fix Coverity issues

AdamZelikNS opened the pull request:
Fix detections with potentially real impact.
Suppress false positives by adding comments: coverity[tag]

The Coverity analysis report can be found here: [Jenkins-run#9](https://jenkins-ncs.nordicsemi.no/blue/organizations/jenkins/latest%2Ftest-fw-nrfconnect-rs_drv154_integration/detail/fix%2FKRKNWK-8993-coverity_fixes_check/9/artifacts/) (no detections)

hubertmis commented on src/nrf_drv_radio802154_rx_buffer.c:
```diff
@@ -44,34 +44,23 @@
-static rx_buffer_t nrf_drv_radio802154_rx_buffers[RADIO_RX_BUFFERS];
-
-#endif
+static rx_buffer_t m_rx_buffers[RADIO_RX_BUFFERS]; ///< Receive buffers.
```
Please, don't rename this variable. It should be `nrf_drv_radio802154_rx_buffers`. And it should not be static. It is available be `extern` in header file.

LuDuda replied:
Ahh okay, i see how it is done for RADIO_RX_BUFFERS = 1.
What about to get rid of extern variable, still keep it a a static and provide getter in nrf_drv_radio802154_rx_buffer.h/c?

hubertmis replied:
Nope. It will provide unnecessary overhead in FSM module in case there is 1 buffer. With `extern` instead of getter we can use `uint8_t * const` buffer type in FSM that is optimized much better than `uint8_t *`.

hubertmis replied:
Unless there is a way to use `uint8_t * const buffer = getter();` or sth similar. I cannot do it :)

LuDuda replied:
Discuss internally to leave extern variable.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver?pullRequest=11) <br/>All committers have signed the CLA.

AdamZelikNS commented:
DNM - Tests on HW not performed yet
(only build)

e-rk commented on src/mac_features/nrf_802154_frame_parser.c:
```diff
@@ -264,19 +264,28 @@ static uint8_t security_offset_get(const uint8_t * p_frame)
+    {
+        return 0;
+    }
+    else
```
nit: since the previous `if` returns, the `else` can be dropped.

AdamZelikNS committed:
Fix Coverity issues

Fix detections with potentially real impact.
Suppress false positives by adding comments: coverity[tag]

Signed-off-by: Adam Zelik <adam.zelik@nordicsemi.no>

e-rk commented on examples/nrf_802154_delayed_echo_client/src/nrf_802154_delayed_echo_client.c:
```diff
@@ -228,6 +228,7 @@ int main(void)
     frame_sequence_number_set(sequence_number_generate());
     nrf_802154_transmit_raw(mp_frame, true);
 
+    // coverity[no_escape]
```
This tag should be added to other examples as well.

hubertmis commented on src/nrf_802154_core.c:
```diff
@@ -2127,6 +2127,7 @@ static bool ack_match_check_version_2(const uint8_t * p_tx_data, const uint8_t *
 
     parse_result = nrf_802154_frame_parser_mhr_parse(p_tx_data, &tx_mhr_data);
     assert(parse_result);
+    (void)parse_result; // suppress false positive of static analysis
```
I'm not sure if this comment is needed. It also suppresses gcc warnings if building with -NDEBUG. Similar phrases like `assert(sth); (void)sth;` is placed all-over the driver code.

jciupis replied:
I'd remove the comments entirely.

hubertmis commented on src/nrf_802154_core.c:
```diff
@@ -2481,6 +2482,7 @@ bool nrf_802154_core_transmit(nrf_802154_term_t              term_lvl,
                 state_set(cca ? RADIO_STATE_CCA_TX : RADIO_STATE_TX);
                 mp_tx_data = p_data;
 
+                // coverity[check_return]
```
I was thinking if instead of suppressing coverity warning in a comment, would it be possible to simply add `(void)result;` before `result = true;` to the `else` block below?

AdamZelikNS replied:
No, it doesn't work here. I have made such attempts before:
- to insert `(void)result;` right after `result = tx_init( . . );` 
- to insert it in `else { (void)result; result = true; }`.
For some unknown reason, neither of the above worked.

hubertmis commented on src/nrf_802154_trx.c:
```diff
@@ -670,6 +670,7 @@ static void rx_antenna_update(void)
     }
 
     assert(result);
+    (void)result; //suppress false positive of static analysis
```
Again, I would add this comment to each similar phrase, or remove it from here

hubertmis reviewed (commented):
I'm looking forward to green coverity run.
I've added some thoughts how to improve this module.
